### ISSUE TITLE
Re-enable cloud/official app

### ISSUE THREAD

**Issue by csu** [fritz-smh/yi-hack#100]
Is there a way to re-enable the official cloud functionality while keeping the HTTP/FTP/RTSP of the custom firmware? Which part of `equip_test.sh` prevents the cloud functionality from starting?

**Comment by notdone** [comment]
Yes, i also just want the index page of HTTP server like this hack when i surf to http://ip cam. 'Cause fw of my cam 1861 cant downgrade to fw 1851. So can i mod the index page of HTTP server (fw 1.8.6.1B)? Pls

**Comment by death2all110** [comment]
@fritz-smh I would like to know this as well. Would it be possible to re-enable the cloud functionality, while keeping the HTTP/FTP portion of the custom firmware?

**Comment by death2all110** [comment]
@csu @notdone I found this here: https://github.com/fritz-smh/yi-hack/issues/26 Looks like if you add back 

```
./p2psever &
./cloud & 
```

after `./mp4record 60 &` on [Line 363 in equip_test.sh](https://github.com/fritz-smh/yi-hack/blob/master/sd/test/equip_test.sh#L363) it should bring back cloud functionality.

**Comment by death2all110** [comment]
Update: Did this on my camera at home. Couple issues. 

1. I noticed my camera frequently disconnects from wifi and needs power cycled to bring back online. 

2. Alerts do not work through the Yi app with this. Not sure why.

**Comment by csu** [comment]
Thanks for testing it out, @death2all110. Breaking the official app alerts is a deal breaker for me :(

**Comment by death2all110** [comment]
@csu Same here. 2 of my cameras are back to stock, I have a 3rd here that I'm tinkering with to see if I can get alert functionality back with the custom fw

**Comment by death2all110** [comment]
@CSU Great news! Got it figured out. Here is an excerpt of my equip_test.sh

```
...
# fix logo
cp -f /home/hd1/test/*.bmp /home/
mv /home/recv.726 /home/recv_X.726

### Key stuff for fixing cloud
if [ -f "/home/notfirst" ]; then         
if [ -f "/home/wpa_supplicant.conf" ]; then         
echo "wifi connectting"
/home/monitor_wifi &
cd /home
else                              
echo "1" > /tmp/waitdump
echo "1" > /tmp/announce_success  
/home/monitor_dump && rm /tmp/waitdump && /home/monitor_wifi &                 
cd /home
fi 
else
echo "first start"
fi 

# do rmm stuff now after calling monitor_wifi, then call the record_event, mp4record, etc
cd /home
./rmm &


### Launch record event
cd /home
./record_event &
./mp4record 60 &
# add cloud and watch_process back.
#./p2pserver &
./cloud &
/home/watch_process &

### Some configuration

himm 0x20050068 0x327c2c
#himm 0x20050068 0x0032562c
...
```

I moved the rmm script execute above the record_event and cloud stuff, and commented out the 
`/home/rmm "/home/hd1/test/voice/success.g726" 1` line below all that.

But I believe the key to it all is the monitor_wifi stuff that is above the rmm and cloud stuff. When checking logs over telnet it was failing with some cloud stuff until I added that back in. Also, the ./p2pserver can be commented out/removed as the cloud script or monitor_wifi seems to call it on their own.

I hope this all makes sense and isn't too confusing!

**Comment by death2all110** [comment]
@csu, @notdone 

I decided to fork @fritz-smh work on this and added the cloud toggle into my own fork.

You can find the commit [here](https://github.com/death2all110/yi-hack/commit/89549759e5fb4167c95efd96d45ee66ddb1033cf)

And you can find my fork [here](https://github.com/death2all110/yi-hack/)
